### PR TITLE
AP_Mount: Improve instance validation check

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Review by amilcarlucas — APPROVED** [review]

**Review by jaxxzer — APPROVED** [review]

**Review by magicrub — APPROVED** [review]

File: libraries/AP_Mount/AP_Mount.cpp [hunk 1]
```diff
@@ -410,6 +410,16 @@ AP_Mount::AP_Mount()
 	AP_Param::setup_object_defaults(this, var_info);
 }
 
+bool AP_Mount::check_primary() const
```
Comment (rmackay9) [inline libraries/AP_Mount/AP_Mount.cpp:413]: Not a big deal but to make it easier for developers to find the code they're looking for I normally try to keep the implementations in the same order as the definitions.  So in this case I'd move these new private method implementations much lower in the .cpp file.  For example they could appear just before handle_gimbal_report.

**Review by rmackay9 — DISMISSED** [review]
small request to move the implementation lower in the .cpp file

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
I'll merge this after the rearrangement.

**Comment by patrickelectric** [comment]
Done

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**